### DISCUSSION TITLE
clipboard not working on OSX 12.2

### DISCUSSION THREAD

**Discussion by rameshsanth** [neovide/neovide#1220] [Q&A] [answered]
I'm using Neovide 0.8.0 . I can't copy or paste to/from clipboard.
Is this not supported or an issue in my setup.

**Comment by dineshKumar777** [comment]
add `set clipboard+=unnamedplus` in your config??
refer `:help clipboard` in nvim

**Reply by rameshsanth** [reply]
i have clipboard=unnamedplus . Doesn't seem to do anything

Cmd-c & Cmd-v don't seem to have any effect at all..

**Comment by bruno-robert** [comment]
setting set clipboard+=unnamedplus essentially makes nvim/vim share it's yank "clipboard" with the systems clipboard.
So when you select some text in nvim/vim and press "y" to yank it, it will also bring it into the clipboard. It doesn't allow you to use the system commands (like CMD+C and CMD+V on macOS)
A lot of actions will now start adding text to your system clipboard (for example "dd" to delete a line).
see [https://stackoverflow.com/a/30691754/7938722](https://stackoverflow.com/a/30691754/7938722)

Other apps like [MacVim](https://macvim-dev.github.io/macvim/) will make use of the system default clipboard shortcut so you can use CMD+C / CTRL+C etc.  



Neovide doesn't seem to support the system shortcuts as of now.

**Comment by bruno-robert** [comment, marked as answer]
Just figured this one out, you can add the following to your `.vimrc` or `init.vim` in order to achieve what you want in neovide.
Enable "logo" key (ie. CMD key on macos).
Map cmd+c etc. to whatever you want!

```
if exists('g:neovide')
  let g:neovide_input_use_logo=v:true
  " copy
  vnoremap <D-c> "+y

  " paste
  nnoremap <D-v> "+p
  inoremap <D-v> <Esc>"+pa
  cnoremap <D-v> <c-r>+

  " undo
  nnoremap <D-z> u
  inoremap <D-z> <Esc>ua
endif
```

Edit: added mapping for command line mode as suggested by @pbnj

**Reply by pbnj** [reply]
And don't forget `cnoremap <D-v> <c-r>+` for command-line mode (i.e. `:{cmd}`)

**Reply by pbnj** [reply]
And here is another one to paste in terminal mode:

In lua (v0.7+):
```lua
vim.api.nvim_set_keymap('t', '<D-v>', '<c-\\><c-n>"+pi<right>', { noremap = true })
```

In vimscript:
```text
tnoremap <D-v> <C-\><C-n>"+pi<right>
```

I don't know if this is the best way, but what it does is:
- `<c-\><c-n>` to switch to normal mode in terminal buffer
- `"+p` to paste from system clipboard
- `i` to switch to insert mode in terminal buffer
- `<right>` to move the cursor 1 character to the right and remove highlighting from the pasted text

**Reply by bruno-robert** [reply]
nice one! you could probably replace the `i<right>` with `a` in order to enter insert mode after the cursor
